AcademicReward/Logic/AddMemberLogic.cs

```diff
@@ -76,5 +70,32 @@ public class AddMemberLogic : ILogic {
         public LogicErrorType AddItem(object obj) {
             return LogicErrorType.NotImplemented;
         }
+
+        /// <summary>
+        /// Helper method used to check the given member username
+        /// </summary>
+        /// <param name="username">string username</param>
+        /// <returns>LogicErrorType logicError</returns>
+        private LogicErrorType AddMemberCheck(string username) {
+            LogicErrorType logicError;
+            if(string.IsNullOrEmpty(username)) {
+                logicError = LogicErrorType.EmptyUsername;
+            } else if(CheckUsernameLength(username)) {
+                logicError = LogicErrorType.InvalidUsernameLength;
+            } else {
+                logicError = LogicErrorType.NoError;
+            }
+            return logicError;
+        }
+
+        /// <summary>
+        /// Helper method used to check the username length
+        /// </summary>
+        /// <param name="username">string username</param>
+        /// <returns>true/false</returns>
+        private bool CheckUsernameLength(string username) {
+            int usernameLength = username.Length;
+            return usernameLength < Profile.MinUsernameLength || usernameLength > Profile.MaxUsernameLength;
```

Comment:
For some reason this throws a warning for me that it will always return false? It does this in other files like LoginLogic.cs

Reply:
I'm not really sure why it would be doing that. If the username length is somehow negative (which I guess that could never happen), return true. If the username is greater than 25, return true. There is definitely a way for this to evaluate to true. I also tested it and it works, same with login and such